x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -62,11 +65,18 @@ public HealthIndicatorResult calculate(boolean includeDetails) {
                 Collections.emptyList()
             );
         } else if (ilmMetadata.getOperationMode() != OperationMode.RUNNING) {
+            List<HealthIndicatorImpact> impacts = Collections.singletonList(
+                new HealthIndicatorImpact(
+                    3,
+                    "Indices are not being rolled over, which could lead to future instability.",
```

Comment:
Is there anything more specific that could be said about "future instability"? As in, could you get more specific around potential for running out of resources or data loss or something that might be coming?

Reply:
Just thinking through what happens when shards get too big:

- Slow query response times
- OutOfMemoryErrors
- Disks filling up (leading to maybe slow query response times at first, server crashes eventually)
- Slow recovery time on node failure (which means data might be under-replicated for longer, so slower query response times).

I'm actually not sure if all of those are still the case in 8.0. I know I had some 1 TB shards in 1.x or 2.x, and the whole cluster was just an unstable mess at that point. Maybe slow query times is the thing to highlight here? @andreidan does that list above look right? Does slow query times seem like the thing to highlight here?

Reply:
I've updated this list based on feedback from @Leaf-Lin at https://github.com/elastic/elasticsearch/pull/86197#pullrequestreview-962768375.

Reply:
++ Keith - I think for a user the "data retention" aspect would be very important (data staying for too long in the system could breach some user data laws and such) and financially, data not moving through the tiers will have a very unpleasant impact on the bills